pdns/pdns_recursor.cc

```diff
@@ -1128,6 +1128,56 @@ int followCNAMERecords(vector<DNSRecord>& ret, const QType& qtype)
   return rcode;
 }
 
+enum class PolicyResult : uint8_t { NoAction, HaveAnswer, Drop };
+
+static PolicyResult handlePolicyHit(const DNSFilterEngine::Policy& appliedPolicy, const std::unique_ptr<DNSComboWriter>& dc, SyncRes& sr, int& res, vector<DNSRecord>& ret, DNSPacketWriter& pw)
```

Comment:
I think the existing code only skipped that for the `NoAction` case of the first block, which is why I skipped calling this function in that exact case?
The `truncate called over UDP` case also changed but I believe the new behavior might be better (we only count Truncate matches over TCP). If anything else changed, it's likely a mistake :)

Reply:
Looking  more closely..

"we only count Truncate matches over TCP". Unless I'm mistaken, the original code count only truncation cases over *UDP*, and with your changes we count them interdependent of type. The third block did that already.

For the second block the increment is now not done for `NoAction` while previously it was. But that is a "can't happen case", so OK.

For the third block it's indeed the same as before.
So on second thought the code is indeed equivalent (except for the TCP case, but I'm fine with that).

Reply:
Oh, yes, sorry, I meant UDP! On second thought I'll amend the code to only account truncation actions over UDP, in order to prevent discrepancies after an upgrade.

Reply:
The first block is also not consistent with the 2nd and third. But I think that truncation is normally handled by the fist block and not later in the process.